### ISSUE TITLE
`dev_set` and `eval_sets` in espnet2

### ISSUE THREAD

**Issue by sw005320** [espnet/espnet#2126]
I've found some issues in the current design for `dev_set` and `eval_sets` in https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/asr1/asr.sh

The role of the development set is different depending on as a part of the test set to tune some search parameters/hyperparameters **after recognition** or monitoring the loss **during the training process** (and also tune the optimizer based on this dev score).
So, this role must be separated depending on the case, and actually we often use different sets for both cases.
For example, in the librispeech recipe, we use the combined and pruned development set during training while we use the original dev set (dev_clean and dev_other) during inference. Note that they are slightly different but mostly the same.
Then, the current script tries to recognize both `dev_set` and `eval_sets` in stage 11.
Since the dev sets used in training and inference are different, we need to add the original dev sets to `eval_sets`, which means that we almost recognize the same dev sets twice. Also, adding the dev sets is inconsistent with the name of the variable `eval_sets`.

My suggestion is to first rename them as follows (basically try to avoid to use the name dev and eval in the variable name level).
- `dev_set` -> `val_set` (or `train_dev` like in espnet1)
- `eval_sets` -> `test_sets` (or `recog_set` like in espnet1)

Also, instead of always recognizing both `dev_set` and `eval_sets`, we only recognize `test_sets`, in stage 11 (https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/asr1/asr.sh#L944) and stage 12 (https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/asr1/asr.sh#L999). Since it can accept multiple data sets, we can include the original development sets in `test_sets`.
Actually, espnet1 is designed like that to deal with such an issue with some consideration of the above multiple dev set roles.
